test/vtgatev3_test.py

```diff
@@ -948,12 +1005,55 @@ def test_main_seq(self):
     
     # Now test direct calls to sequence.
     result = self.execute_on_master(
-        vtgate_conn, "select next 1 values from vt_main_seq", {})
+        vtgate_conn, 'select next 1 values from vt_main_seq', {})
     self.assertEqual(
         result,
         ([(5,)], 1, 0,
          [('nextval', self.int_type)]))
 
+  def test_upsert(self):
+    vtgate_conn = get_connection()
+
+    # Create lookup entries for primary vindexx:
```

Comment:
`upsert_primary` is not owned by this table, which means that 'someone else' has to create entries prior to the insert. So, we pre-create those rows here. I've added a comment to elaborate.

`upsert_owned` is owned. So, an insert will cause rows to be created there.

The test is basically testing behavior for both owned and unowned vindexes.